entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityTypeHandler.kt

```diff
@@ -3,23 +3,48 @@ package com.egm.stellio.entity.web
 import com.egm.stellio.entity.service.EntityTypeService
 import com.egm.stellio.entity.util.decode
 import com.egm.stellio.shared.model.ResourceNotFoundException
-import com.egm.stellio.shared.util.*
+import com.egm.stellio.shared.util.JSON_LD_CONTENT_TYPE
 import com.egm.stellio.shared.util.JsonLdUtils.expandJsonLdKey
+import com.egm.stellio.shared.util.JsonUtils
+import com.egm.stellio.shared.util.buildGetSuccessResponse
+import com.egm.stellio.shared.util.getContextFromLinkHeaderOrDefault
 import org.springframework.http.HttpHeaders
 import org.springframework.http.MediaType
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
+import java.util.Optional
 
 @RestController
 @RequestMapping("/ngsi-ld/v1/types")
 class EntityTypeHandler(
     private val entityTypeService: EntityTypeService
 ) {
 
+    /**
+     * Implements 6.25 - Retrieve Available Entity Types
+     */
+    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun getTypes(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @RequestParam details: Optional<Boolean>
+    ): ResponseEntity<*> {
+        val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
+        val detailedRepresentation = details.orElse(false)
+
+        val availableEntityTypes: Any = if (detailedRepresentation)
+            entityTypeService.getEntityTypes(listOf(contextLink))
+        else
+            entityTypeService.getEntityTypeList(listOf(contextLink))
```

Comment:
if not in detailed representation, you never compact the entity types, do you?

Reply:
Indeed, they should be returned compacted, fixed.